datafusion/physical-plan/src/aggregates/mod.rs

```diff
@@ -751,28 +771,16 @@ impl AggregateExec {
                 })
             }
             _ => {
-                // When the input row count is 0 or 1, we can adopt that statistic keeping its reliability.
+                // When the input row count is 1, we can adopt that statistic keeping its reliability.
                 // When it is larger than 1, we degrade the precision since it may decrease after aggregation.
-                let num_rows = if let Some(value) = self
-                    .input()
-                    .partition_statistics(None)?
-                    .num_rows
-                    .get_value()
+                let num_rows = if let Some(value) = child_statistics.num_rows.get_value()
                 {
-                    if *value > 1 {
-                        self.input()
-                            .partition_statistics(None)?
-                            .num_rows
-                            .to_inexact()
-                    } else if *value == 0 {
-                        // Aggregation on an empty table creates a null row.
```

Comment:
*   If `!group_by_expr.is_empty()` and `input_statistics.num_rows == 0`:
    *   Both `Partial` and `Final` aggregation modes (`agg.mode`) correctly yield 0 output rows.
    *   I added `dbg!(&output_batch.num_rows())` before [this line](https://github.com/apache/datafusion/blob/9d2f04996604e709ee440b65f41e7b882f50b788/datafusion/physical-plan/src/aggregates/row_hash.rs#L800). It printed nothing, since `self.exec_state` likely transitioned to `ExecutionState::Done`.
*   If `group_by_expr.is_empty()` and `input_statistics.num_rows == 0`:
    *   `Final` aggregation mode (`agg.mode == Final`) yields 1 output row

Reply:
Does it seem like a bug fix? If so, it's better add a unit test to cover it

Reply:
I think current unit test already covered it 
+ [`group_by_expr.is_empty() and input_statistics.num_rows == 0`](https://github.com/apache/datafusion/blob/0c7760aebd4e057196b363a173a9884bcdbb7236/datafusion/core/tests/physical_optimizer/partition_statistics.rs#L673-L708)
+ [`!group_by_expr.is_empty() and input_statistics.num_rows == 0`](https://github.com/apache/datafusion/blob/0c7760aebd4e057196b363a173a9884bcdbb7236/datafusion/core/tests/physical_optimizer/partition_statistics.rs#L632-L663)